language/move-vm/runtime/src/loader.rs

```diff
@@ -981,17 +985,18 @@ impl Loader {
 
             Type::Vector(ty) => Ok(AbilitySet::polymorphic_abilities(
                 AbilitySet::VECTOR,
-                vec![self.abilities(ty)?].into_iter(),
+                vec![(self.abilities(ty)?, false)].into_iter(),
             )),
             Type::Struct(idx) => Ok(self.module_cache.read().struct_at(*idx).abilities),
             Type::StructInstantiation(idx, type_args) => {
-                let declared_abilities = self.module_cache.read().struct_at(*idx).abilities;
+                let struct_type = self.module_cache.read().struct_at(*idx);
                 let type_argument_abilities = type_args
```

Comment:
wild, didn't notice before. Or I did and looked it up then. But do we check some where that the correct number of type arguments were passed in?

Reply:
None of the code I touched seems to check it, but there may be a check somewhere else? I just went on and zip the args and params.